packages/projects/src/components/ProjectForm.jsx

```diff
@@ -107,39 +119,57 @@ const toKebabCase = str =>
       .join('-')
       .toLowerCase()
 
+const setUrl = async (state, dispatch) => {
+  const title = value(state, 'title')
+  const url = value(state, 'url')
+
+  if (!url || url.length === 0) {
+    const suggestedUrl = toKebabCase(title)
+    dispatch({ type: 'urlStatus', payload: URL_STATUSES.clean })
+    dispatch({ type: 'url', payload: suggestedUrl })
+    await validateUrl(suggestedUrl, state, dispatch)
+  }
+}
+
 const setUrlPlaceholder = (title, dispatch) => {
   const newUrlPlaceholder = toKebabCase(title)
   dispatch({ type: 'urlPlaceholder', payload: newUrlPlaceholder })
 }
 
-const validateUrl = async (state, dispatch, e) => {
+const onUrlBlur = (state, dispatch, e) => {
   e.preventDefault()
 
   const url = e.target.value
 
-  let urlStatus
+  validateUrl(url, state, dispatch)
```

Comment:
nit: you should `await`.

while this works as is right now, it's going to cause an issue in the future if you extend the function.